pygfx/utils/__init__.py

```diff
@@ -219,3 +219,43 @@ def assert_type(name, value, *classes):
 
         # Raise message with alt traceback
         raise TypeError(msg).with_traceback(tb) from None
+
+
+class ReadOnlyDict(dict):
```

Comment:
I wonder if it would be more performant to wrap a dict instead of subclassing?

Reply:
I want it to be a dict, e.g. with `isinstance()`.